ps-menu.psm1

```diff
@@ -1,87 +1,122 @@
-function DrawMenu {
+write-debug "Loading: ps-menu.psm1"
+
+function DrawMenu
+{
+    [cmdletbinding()]
     param ($menuItems, $menuPosition, $Multiselect, $selection)
+
+	# ensure preferences passed down to continue into function
     $l = $menuItems.length
-    for ($i = 0; $i -le $l;$i++) {
-		if ($menuItems[$i] -ne $null){
-			$item = $menuItems[$i]
-			if ($Multiselect)
-			{
-				if ($selection -contains $i){
-					$item = '[x] ' + $item
-				}
-				else {
-					$item = '[ ] ' + $item
-				}
-			}
-			if ($i -eq $menuPosition) {
-				Write-Host "> $($item)" -ForegroundColor Green
-			} else {
-				Write-Host "  $($item)"
-			}
-		}
+    for ($i = 0; $i -le $l; $i++)
```

Comment:
I see a bug here, there should be: `$i -lt $l` instead (lower than vs lower than or equal)